lib/src/modules/compiler/template_parser/GRAMMAR.md

```diff
@@ -0,0 +1,38 @@
+#Template Grammar
+
+###Todos
+* Can you interpolate comments?
+* Can you have multiple pipes?
+* What is the best way to handle the additional constraints on expressions?
```

Comment:
Good question: I don't know how much of the Dart language AST we are going to need to support "full" expressions. Maybe it's worth getting a copy of the Dart expression grammar to see what's worth supporting.